core/src/main/java/io/kestra/core/tasks/flows/ForEachItem.java

```diff
@@ -0,0 +1,278 @@
+import io.kestra.core.models.flows.Flow;
+import io.kestra.core.models.tasks.ExecutableTask;
+import io.kestra.core.models.tasks.Task;
+import io.kestra.core.runners.ExecutableUtils;
+import io.kestra.core.runners.FlowExecutorInterface;
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.runners.WorkerTaskExecution;
+import io.kestra.core.runners.WorkerTaskResult;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import lombok.experimental.SuperBuilder;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Positive;
+
+import static io.kestra.core.utils.Rethrow.throwFunction;
+
+@SuperBuilder(toBuilder = true)
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Execute a subflow for each batch of items",
+    description = "Execute a subflow for each batch of items. The `items` value must be internal storage URI e.g. an output file from a previous task, or a file from inputs of FILE type."
+)
+@Plugin(
+    examples = {
+        @Example(
+            title = "Execute a subflow for each batch of items",
+            code = {
+                """
+                    id: each
+                    type: io.kestra.core.tasks.flows.ForEachItem
+                    items: "{{ outputs.extract.uri }}" # works with API payloads too. Kestra can detect if this output is not a file,\s
+                    # and will make it to a file, split into (batches of) items
+                    maxItemsPerBatch: 10
+                    subflow:
+                      flowId: file
+                      namespace: dev
+                      inputs:
+                        file: "{{ taskrun.items }}" # special variable that contains the items of the batch
+                      wait: true # wait for the subflow execution
+                      transmitFailed: true # fail the task run if the subflow execution fails"""
+            }
+        )
+    }
+)
+public class ForEachItem extends Task implements ExecutableTask {
+    private static final String URI_FORMAT = "kestra:///%s/%s/executions/%s/tasks/%s/%s/batch-%s.ion";
+
+    @NotEmpty
+    @PluginProperty(dynamic = true)
+    @Schema(title = "The items to be split into batches and processed. Make sure to set it to Kestra's internal storage URI, e.g. output from a previous task in the format `{{ outputs.task_id.uri }}` or an input parameter of FILE type e.g. `{{ inputs.myfile }}`.")
```

Comment:
Or from the namespace files storage ?

Reply:
This is the standard comment, but yes, it should be possible. But note that we currently have no way to have the storage URI of a namespace file easily (we only have a Pebble function to read the content).
For now, I prefer to keep the standard comment we use everywhere